### ISSUE TITLE
`Style/TopLevelMethodDefinition` false positives

### ISSUE THREAD

**Issue by mockdeep** [rubocop/rubocop#10799]
I'm seeing a lot of false positives on the `Style/TopLevelMethodDefinition` cop when upgrading to version `1.31.2`. It seems to be complaining about any top level method *call*.

## Expected behavior

It should not report issues in the scenarios below.

## Actual behavior

It reports errors on top-level method calls.

## Steps to reproduce the problem

```rb
require "action_cable"
```

**Output**

```
app/channels/application_cable/channel.rb:1:1: C: Style/TopLevelMethodDefinition: Do not define methods at the top-level.
```

It also, oddly, reports on static HAML files when running with `haml-lint`:

```haml
%h1 New Account
```

**Output**

```
app/views/accounts/new.html.haml:1 [W] RuboCop: Style/TopLevelMethodDefinition: Do not define methods at the top-level.
```

You can see the issue on [this PR](https://github.com/mockdeep/lmkw/pull/730) ([build here](https://app.circleci.com/pipelines/github/mockdeep/lmkw/2136/workflows/4471e452-8252-4a0d-8a09-0c5e10ee2e10/jobs/2152))

## RuboCop version

```
1.31.2 (using Parser 3.1.2.0, rubocop-ast 1.18.0, running on ruby 3.1.2 x86_64-linux)
  - rubocop-rails 2.15.2
  - rubocop-rspec 2.12.1
```

**Comment by kachick** [comment]
Looks same of https://github.com/rubocop/rubocop/pull/10796 👀

**Comment by mockdeep** [comment]
Ah, yep! I looked at the issues and not the PRs.
